index.js

```diff
@@ -42,23 +41,38 @@ const setRetrievalResult = async (req, res, client, retrievalId) => {
   assert(!Number.isNaN(retrievalId), 400, 'Invalid Retrieval ID')
   const body = await getRawBody(req, { limit: '100kb' })
   const result = JSON.parse(body)
+  assert.strictEqual(typeof result.walletAddress, 'string', 400, 'Invalid .walletAddress')
+  assert.strictEqual(typeof result.success, 'boolean', 400, 'Invalid .success')
+  assert.strictEqual(typeof result.startAt, 'string', 400, 'Invalid .startAt')
+  assert.strictEqual(typeof result.statusCode, 'number', 400, 'Invalid .statusCode')
+  assert.strictEqual(typeof result.firstByteAt, 'string', 400, 'Invalid .firstByteAt')
+  assert.strictEqual(typeof result.endAt, 'string', 400, 'Invalid .endAt')
+  assert.strictEqual(typeof result.byteLength, 'number', 400, 'Invalid .byteLength')
   try {
     await client.query(`
-      INSERT INTO retrieval_results
-      (retrieval_id, wallet_address, success)
-      VALUES ($1, $2, $3)
+      INSERT INTO retrieval_results (
+        retrieval_id, wallet_address, success, start_at, status_code,
+        first_byte_at, end_at, byte_length
+      )
+      VALUES (
+        $1, $2, $3, to_timestamp($4), $5, to_timestamp($6), to_timestamp($7),
+        $8
+      )
     `, [
       retrievalId,
       result.walletAddress,
-      result.success
+      result.success,
+      new Date(result.startAt).getTime(),
+      result.statusCode,
+      new Date(result.firstByteAt).getTime(),
+      new Date(result.endAt).getTime(),
```

Comment:
I think you should validate that the date values are correctly formatted and return 400 error response if they are not.

Please include tests for this validation.

```
❯ node
Welcome to Node.js v18.14.2.
Type ".help" for more information.
> new Date('not-iso').getTime()
NaN
```